nc-news/src/components/TopicBody.js

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react"
+import {
+    Routes,
+    Route,
+    Link,
+} from 'react-router-dom';
+import { fetchTopics } from "../api";
+import ArticleList from "./ArticleList";
+import TopicCard from "./TopicCard";
+
+export default function TopicBody() {
+
+    const [topicsState, setTopicsState] = useState()
+    const [loadingState, setLoadingState] = useState(true);
+
+    useEffect(() => {
+        fetchTopics()
+            .then((topics) => {
+                setTopicsState(topics);
+                setLoadingState(false);
+            })
+    },
+        [])
+
+    if (loadingState) {
+        return <p>Loading...</p>
+    } else {
+        return (
+            <section>
+                <h2>Topics</h2>
+                <section className="TopicBody-TopicTiles">{topicsState.map(topic => {
+                    const path = `/topics/${topic.slug}`
+                    return (
+                        <Link key={topic.slug} className="TopicCard" to={path}>
+                            <TopicCard topic={topic} />
+                        </Link>
+                    )
+                })}</section>
+                <section>
+                    <Routes>
+                        <Route path='/:topic' element={<ArticleList />} />
```

Comment:
Ideally you want to declare routes inside App.js. Doing it this way could get messy real quick.

Reply:
Hi Karla,


I spoke through with Jim and he mentioned the same thing. He did say I'd be alright to keep sub-routing now that it's there (and that it was one way to ensure that I could continue to render my topic tiles whilst displaying the relevant articles.

If it's okay I'll keep subrouting in, if only to remind me how it works when I look back over my work!


Thanks,

Callum